deploy.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+
+UPDATE=0
+DEST_DIR=""
+FED_KEY=""
+MAIN_BLOCK_NUMBER=""
+SIDE_BLOCK_NUMBER=""
+ETH_HOST=""
+RSK_HOST=""
+PROGRAMS="docker npm nodejs jq"
+
+quit() {
+    if [ $1 -eq 1 ]; then
+        echo "ERROR: "$2
+    else
+        echo $2
+    fi
+    exit $1
+}
+
+config() {
+    cp $DEST_DIR/federator/config/config.sample.js $DEST_DIR/federator/config/config.js &&
+        sed -i "s|rsktestnet-kovan|rskmainnet|g" $DEST_DIR/federator/config/config.js &&
+        sed -i "s|kovan|ethmainnet|g" $DEST_DIR/federator/config/config.js &&
+        return 0
+    quit 1 "There was a problem setting the configuration, please verify."
+}
+
+check_required_programs() {
+    echo "Checking for required programs..."
+    rc=0
+    PROGRAMS=$1
+    for program in $PROGRAMS; do
+        if ! command -v "$program" >/dev/null 2>&1; then
+            rc=1
+            echo "$program: command not found"
+        fi
+    done
+    if [ $rc -ne 0 ]; then
+        quit 1 "Requirements not acomplished"
+    fi
+}
+
+dependencies() {
+    echo "Installing required packages ..."
+    npm install &&
+        echo "Packages installed" &&
+        return 0
+    quit 1 "There was an error installing the needed packages, please check"
+}
+
+key() {
+    while [ ! -f $DEST_DIR/federator/config/federator.key ]; do
+        read -p "Please move your private key to $DEST_DIR/federator/config/federator.key, then press any key to continue " FED_KEY
+        continue
+    done
+    echo ""
+    echo "Private key placed on $DEST_DIR/federator/config/federator.key"
+    return 0
+}
+
+block_exists() {
+    grep -q '^[0-9]*$' "$DEST_DIR/federator/db/lastBlock.txt" 2>/dev/null &&
+        grep -q '^[0-9]*$' "$DEST_DIR/federator/db/side-fed/lastBlock.txt" 2>/dev/null &&
+        return 0
+    return 1
+}
+
+eth_host(){
+    read -p "Please enter your ETH host address along with the RPC port, (https://127.0.0.1:8545) " ETH_HOST
+    [ ! -z "${ETH_HOST}" ] &&
+        sed -i "s|<YOUR HOST URL AND PORT>|$ETH_HOST|g" $DEST_DIR/federator/config/ethmainnet.json &&
+        return 0
+    quit 1 "There was an error setting the ETH host and port"
+}
+
+rsk_host(){
+    read -p "Please enter your RSK host address along with the RPC port, (https://127.0.0.1:4444) " RSK_HOST
+    [ ! -z "${RSK_HOST}" ] &&
+        sed -i "s|<YOUR HOST URL AND PORT>|$RSK_HOST|g" $DEST_DIR/federator/config/rskmainnet.json &&
+        return 0
+    quit 1 "There was an error setting the RSK host and port"
+}
+
+last_block_rsk() {
+    LAST_BLOCK_RSK=$(curl -s 'https://backend.explorer.rsk.co/api?module=blocks&action=getBlocks&limit=1')
+    MAIN_BLOCK_NUMBER=$(echo $LAST_BLOCK_RSK | jq '. | .data[0].number')
+}
+
+last_block_eth() {
+    LAST_BLOCK_ETH=$(curl -s https://api.blockcypher.com/v1/eth/main)
+    SIDE_BLOCK_NUMBER=$(echo $LAST_BLOCK_ETH | jq '. | .height')
+}
+
+block() {
+    last_block_eth &&
+        last_block_rsk &&
+        echo $MAIN_BLOCK_NUMBER
```

Comment:
Why? Also you either missed a `&&` at the end, or the next line shouldn't be indented.

Reply:
A debug echo that went through. Good catch